File: app/src/main/java/michael/com/kotlindemo/java/ui/TaskContract.java
package michael.com.kotlindemo.java.ui;

import java.util.List;

import michael.com.kotlindemo.java.model.Task;

/**
 * Created by Mikhail on 5/31/17.
 */

public interface TaskContract {

    interface View {

        void showTasks(List<Task> tasks);

        void showAddedTasks(Task task);

        void showCompletedTasks(List<Task> tasks);

        void showUpdatedTask();

        void showProgressBar();

        void removeProgressBar();

        void showError(String errorMessage);

        void showSuccess(String successMessage);

        void showHeader(String header);

        void clearInputText();

    }

    interface Presenter {

        void loadTasks(boolean isLoading);

        void loadCompletedTasks();

        void saveTask(String text);

        void updateTask(String id, Task task);

        void subscribe();

        void unSubscribe();
    }
}

File: app/src/main/java/michael/com/kotlindemo/kotlin/ui/TaskContract.kt
package michael.com.kotlindemo.kotlin.ui

import michael.com.kotlindemo.kotlin.model.Task

/**
 * Created by Mikhail on 5/31/17.
 */

interface TaskContract {

    interface View {

        fun showTasks(tasks: MutableList<Task>)

        fun showAddedTasks(task: Task)

        fun showCompletedTasks(tasks: MutableList<Task>)

        fun showUpdatedTask()

        fun showProgressBar()

        fun removeProgressBar()

        fun showError(errorMessage: String)

        fun showSuccess(successMessage: String)

        fun showHeader(header: String)

        fun clearInputText()

    }

    interface Presenter {

        fun loadTasks(isLoading: Boolean)

        fun loadCompletedTasks()

        fun saveTask(text: String)

        fun updateTask(id: String, task: Task)

        fun subscribe()

        fun unSubscribe()
    }
}

